src/views/Necessitous/Demand.vue

```diff
@@ -50,6 +96,22 @@ export default class NecessitousDemand extends Vue {
         return this.step();
     }
 
+    private updateSupplies(type: keyof Step.Supplies, position: any) {
+        pipe(
+            O.fromNullable(this.demand.supplies),
+            O.map(supplies => supplies[type]),
+            O.map((supply: any) => {
+                supply.positions = supply.positions
+                    .filter((item: any) =>
+                        item.style === position.style && item.type === position.type ? false : true
+                    )
+                    .filter((item: any) => item.quantity !== 0);
+
+                supply.positions.push(position);
```

Comment:
Mutations in the `map` are breaking the `Functor`'s law. 
But let's keep it as is for now. 🔥